src/disassembling/rzil.md

```diff
@@ -0,0 +1,151 @@
+# RzIL
+
+RzIL is the new intermediate language in Rizin, primarily intended for representing the semantics of machine code. It is designed as a clone of BAP's [Core Theory](http://binaryanalysisplatform.github.io/bap/api/master/bap-core-theory/Bap_core_theory/), with minor deviations where necessary.
+
+More details related the implementation can be found [here](https://github.com/rizinorg/rizin/blob/dev/doc/rzil.md).
+
+## IL statements
+
+IL statements for each [opcode](https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_il/rz_il_opcodes.h) can be represented as a string, composed of s-expressions via command `aoi` (JSON output is available via `aoj`).
+
+In the below shows an example using `aoip` (the *prettified* output) of 2 PPC instructions (`stwu` and `mflr`).
+
+```bash
+[0x10000488]> pd 2
+|           0x10000488      stwu  r1, -0x10(r1)
+|           0x1000048c      mflr  r0
+[0x10000488]> aoi?
+Usage: aoi[p]   # Print the RzIL of next N instructions
+| aoi [<n_instructions>]  # Print the RzIL of next N instructions
+| aoip [<n_instructions>] # Pretty print the RzIL of next N instructions
+[0x10000488]> aoip 2
+0x10000488
+(seq
+  (storew 0
+    (+
+      (var r1)
+      (let v
+        (bv 16 0xfff0)
+        (ite
+          (msb
+            (var v))
+          (cast 32
+            (msb
+              (var v))
+            (var v))
+          (cast 32
+            false
+            (var v)))))
+    (cast 32
+      false
+      (var r1)))
+  (set r1
+    (+
+      (var r1)
+      (let v
+        (bv 16 0xfff0)
+        (ite
+          (msb
+            (var v))
+          (cast 32
+            (msb
+              (var v))
+            (var v))
+          (cast 32
+            false
+            (var v)))))))
+0x1000048c
+(set r0
+  (cast 32
+    false
+    (var lr)))
+```
+
+The same output of `aoi` can be obtained via `rz-asm` as in the following example.
+
+```bash
+$ rz-asm -de -a ppc 7c0802a6
+mflr r0
+$ rz-asm -Ie -a ppc 7c0802a6
+(set r0 (cast 32 false (var lr)))
+```
+
```

Comment:
Please add that it's possible to print RzIL for the whole function using `plf` command.

Reply:
oh, i missed this command.